src/genie/libs/parser/iosxe/show_gnxi.py

```diff
@@ -0,0 +1,288 @@
+                        "oper_up": bool
+                        },
+                    "cert_mgmt_svc": {
+                        "admin_enabled": bool,
+                        "oper_up": bool
+                        },
+                    "os_image_svc": {
+                        "admin_enabled": bool,
+                        "oper_up": bool,
+                        "supported": bool
+                        },
+                    "factory_reset_svc": {
+                        "admin_enabled": bool,
+                        "oper_up": bool,
+                        "supported": bool
+                        }
+                }
+        }
+
+
+class ShowGnxiStateDetail(ShowGnxiStateDetailSchema):
+    ''' Parser for:
+            show gnxi state detail
+    '''
+    cli_command = "show gnxi state detail"
+
+    class GnmibSection(Enum):
+        SETTINGS = "settings"
+        GNMI_GRPC = "grpc"
+        GNMI_CONF = "config_svc"
+        GNMI_TELEMETRY = "telemetry_svc"
+        GNOI_CERT = "cert_mgmt_svc"
+        GNOI_OS = "os_image_svc"
+        GNOI_RESET = "factory_reset_svc"
+
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        current_gnmib_section = self.GnmibSection.SETTINGS
+
+        p_server = re.compile(r"Server: (?P<insecure_server>Disabled|Enabled)")
+        p_server_port = re.compile(r"Server port: (?P<insecure_port>\d+)")
+        p_secure_server = re.compile(r"Secure server: (?P<secure_server>Disabled|Enabled)")
+        p_secure_server_port = re.compile(r"Secure server port: (?P<secure_port>\d+)")
+        p_secure_client_auth = re.compile(r"Secure client authentication: (?P<secure_client_auth>Disabled|Enabled)")
+        p_secure_tp = re.compile(r"Secure trustpoint:\s?(?P<secure_trustpoint>.*)")
+        p_secure_client_tp = re.compile(r"Secure client trustpoint:\s?(?P<secure_client_trustpoint>.*)")
+        p_secure_pwd_auth = re.compile(r"Secure password authentication: (?P<secure_pass_auth>Disabled|Enabled)")
+
+        p_bs_state = re.compile(r"State: (?P<bootstrapping_state>Provisioned|Default)")
+
+        p_section_gnmi = re.compile(r"GNMI")
+        p_grpc_section = re.compile(r"gRPC Server")
+        p_conf_section = re.compile(r"Configuration service")
+        p_telemetry_section = re.compile(r"Telemetry service")
+        p_gnoi_section = re.compile(r"GNOI")
+        p_cert_section = re.compile(r"Cert Management service")
+        p_os_section = re.compile(r"OS Image service")
+        p_reset_section = re.compile(r"Factory Reset service")
+
+        p_admin_state = re.compile(r"Admin state: (?P<admin_state>Disabled|Enabled)")
+        p_oper_status = re.compile(r"Oper status: (?P<oper_status>Down|Up)")
+        p_supported = re.compile(r"Supported: (?P<supported>Not supported on this platform|Supported)")
+
+        ret_dict = defaultdict(dict)
```

Comment:
Using a default dictionary makes it easier to fill in the dictionary since I don't have to insert `.set_default` everywhere I access it. Is there any particular reason that I should avoid using it?

Reply:
for consistency all parsers are dict.

Reply:
The dictionary is passed through various checks and balances behind the scenes. Making it a `defaultdict` has the potential of causing issues down the line. We would rather avoid that possibility entirely

Reply:
I agree. I've changed the return statement to convert `ret_dict` to a dictionary